src/main/java/hu/wurfel/new_school_reference/diary/DiaryService.java

```diff
@@ -14,34 +17,145 @@
 @Service
 public class DiaryService extends CrudService<Diary, DiaryRepository> {
 
-@Autowired
+	private ClassService classService;
+	private TeacherService teacherService;
+
+	@Autowired
 	public DiaryService(DiaryRepository repository) {
 		super(repository);
 	}
 
-	public List<Diary> findAllByStart(Date date){
-	return repository.findAllByStart(date);
+	//region withDiary
+
+	public List<Diary> findAllDiaryByDiaryIfPossible(DiaryDto diaryDto) {
+		if (diaryDto.getId() != 0) {
+			return this.getList(this.findById(diaryDto.getId()));
+		}
+		return findDiaryByDiaryStartIfPossible(diaryDto);
+	}
+
+	private List<Diary> findDiaryByDiaryStartIfPossible(DiaryDto diaryDto) {
+		if (diaryDto.getStart() != null) {
+			return this.findAllByStart(diaryDto.getStart());
+		}
+		return findDiaryByDiaryHeadTeacherIfPossible(diaryDto);
+	}
+
+	private List<Diary> findDiaryByDiaryHeadTeacherIfPossible(DiaryDto diaryDto) {
+		if (diaryDto.getHeadTeacherId() != 0) {
+			return this.findAllByHeadTeacherId(diaryDto.getHeadTeacherId());
+		}
+		return new ArrayList<Diary>();
+	}
+
+	public List<Diary> findAllByStart(Date date) {
+		return repo.findAllByStart(date);
+	}
+
+	public List<Diary> findAllByEnd(Date date) {
+		return repo.findAllByEnd(date);
 	}
 
-	public List<Diary> findAllByEnd(Date date){
-	return repository.findAllByEnd(date);
+
+	public List<Diary> findAllByHeadTeacher(Teacher teacher) {
+		return repo.findAllByHeadTeacher(teacher);
+	}
+
+	public List<Diary> findAllByHeadTeacherId(long teacherId) {
+		return repo.findAllByHeadTeacher(teacherId);
+	}
```

Comment:
Nem tudom, hogy pontosan mi a különbség a kettő között db-ben, de egészen biztos, hogy a teacheres költségesebb. Vagy ha nem, akkor ua. Az id-s tökéletesen egyértelmű és elég is szerintem.

Reply:
done